app/models/ems_event/automate.rb

```diff
@@ -12,6 +12,16 @@ def refresh(*targets, sync)
       EmsRefresh.queue_refresh(refresh_targets, nil, sync)
     end
 
+    def refresh_new_target
+      ems = ext_management_system
+      if ems.supports_refresh_new_target?
+        vm_class = ems.class::Vm
+        target = vm_class.create_from_event(self)
```

Comment:
Okay making progress :)  So this is still creating the VM record in the `Event Hander`, the next step is to put an item on the queue for the `Refresh Worker` that way we don't have any race conditions with two workers creating a record for the same VM.